community-tutorials/assign-additional-ip-to-lxc-container/01-en.md

```diff
@@ -0,0 +1,490 @@
+
+# Step 7 - Installing a TURN server in another container
+
+In some network restricted sites or development environments, such as
+those behind NAT or a firewall that restricts outgoing UDP
+connections, users may be unable to make outgoing UDP connections to
+your BigBlueButton server.
+
+The TURN protocol is designed to allow UDP-based communication flows
+like WebRTC to bypass NAT or firewalls by having the client connect to
+the TURN server, and then have the TURN server connect to the
+destination on their behalf.
+
+In addition, the TURN server implements the STUN protocol as well,
+used to allow direct UDP connections through certain types of
+firewalls which otherwise might not work.
+
+Using a TURN server under your control improves the success of
+connections to BigBlueButton and also improves user privacy, since
+they will no longer be sending IP address information to a public STUN
+server.
+
+Because the TURN protocol is not CPU and memory intensive, and because
+it needs to use the port 443, it makes sense to use another public IP
+for it and to install it in a container with a routed NIC.
+
+## Create the profile
+
+We can copy and modify the profile for BBB:
+
+```bash
+lxc profile copy bbb turn
+lxc profile ls
+lxc profile edit turn
+```
+
+Change the public IP and the `host_name`. It should look like this:
+
+```yaml
+config:
+  security.nesting: "true"
+  user.network-config: |
+    version: 2
+    ethernets:
+        eth0:
+            addresses:
+            - 37.121.183.102/32
+            nameservers:
+                addresses:
+                - 8.8.8.8
+                search: []
+            routes:
+            -   to: 0.0.0.0/0
+                via: 169.254.0.1
+                on-link: true
+description: Routed LXD profile
+devices:
+  eth0:
+    host_name: veth-turn
+    ipv4.address: 37.121.183.102
+    nictype: routed
+    parent: eth0
+    type: nic
+name: turn
+used_by:
+```
+
+The modifications are these:
+
+```yaml
+config:
+    ethernets:
+        eth0:
+            addresses:
+            - 37.121.183.102/32
+
+devices:
+  eth0:
+    host_name: veth-turn
+    ipv4.address: 37.121.183.102
+```
+
+The setting : **`security.nesting: "true"`** is not actually needed
+because we don't need to run docker inside the container, but it
+doesn't harm.
+
+## Launch the container
+
+```bash
+lxc launch ubuntu:20.04 turn --profile default --profile turn
+lxc list
+lxc info turn
+
+ip addr show veth-turn
+ip ro | grep veth
+
+lxc exec turn -- ip addr
+lxc exec turn -- ip ro
+```
+
+### Fix networking
+
+```bash
+firewall-cmd --permanent --zone=trusted --add-interface=veth-turn 
+
+firewall-cmd --permanent --direct --add-rule \
+        ipv4 filter FORWARD 0 -o veth-turn -j ACCEPT
+firewall-cmd --permanent --direct --add-rule \
+        ipv6 filter FORWARD 0 -o veth-turn -j ACCEPT
+
+firewall-cmd --reload
+
+firewall-cmd --zone=trusted --list-all
+iptables-save | grep veth
+
+lxc exec turn -- ping 169.254.0.1
```

Comment:
I can ping with the turn container, but the bbb container is not able to ping anymore. Could it be due to one of the firewall rules?

Reply:
Could it be that you have made some typo or mistake?

Reply:
I don´t think so. I just copied and pasted command by command.

Reply:
Just copy-paste is not good enough. You also have to understand what you are doing.

The difficulty level of this tutorial is "advanced", it is not for novices.
Could you please ask someone who is a bit more experienced than yourself to review it?

Thanks for your understanding.

Reply:
Alright, no problem. There will be another staff member taking over the tutorial in the next few days.